File: client/app/src/main/java/com/qubacy/geoqq/domain/mate/_common/model/chat/MateChat.kt
package com.qubacy.geoqq.domain.mate._common.model.chat

import com.qubacy.geoqq.data.mate.chat.model.DataMateChat
import com.qubacy.geoqq.domain._common.model.user.User
import com.qubacy.geoqq.domain._common.model.user.toUser
import com.qubacy.geoqq.domain.mate._common.model.message.MateMessage
import com.qubacy.geoqq.domain.mate._common.model.message.toMateMessage

data class MateChat(
    val id: Long,
    val user: User,
    val newMessageCount: Int,
    val lastMessage: MateMessage?,
    val lastActionTime: Long
) {

}

fun DataMateChat.toMateChat(): MateChat {
    return MateChat(id, user.toUser(), newMessageCount, lastMessage?.toMateMessage(), lastActionTime)
}

File: server/own_services/common/pkg/domain/geoqq/mateChat.go
package geoqq

import (
	"time"
)

type MateChat struct {
	Id     uint64
	UserId uint64

	NewMessageCount uint64
	LastMessage     *MateMessage
	LastActionTime  time.Time
}

func NewEmptyMateChat() *MateChat {
	return &MateChat{
		0, 0, 0, nil,
		time.Time{}, // ?
	}
}

type MateChatList []*MateChat
